### DISCUSSION TITLE
How to trigger automation when alarm sounds on Homebase

### DISCUSSION THREAD

**Discussion by jamiergrs** [fuatakgun/eufy_security#395] [Q&A]
Hi, I have been trying to work out how I can get an automation to trigger when the home base alarm begins to sound. I have a couple of sensors that have an arrival delay and some that would sound the alarm instantly. 

I can use the Device trigger on the home base to be Base Triggered but that makes the automation run instantly and also when the device is arming with a delay. 

Am I missing something simple that will only trigger if the actual alarm sounds?

**Comment by samturner3** [comment]
I also have this question. Looking at list of states in HA for entity `alarm_control_panel.homebase`:

```
currentMode:
...
  states:
    '0': Away
    '1': Home
    '3': Custom 1
    '4': Custom 2
    '5': Custom 3
    '6': 'Off'
    '63': Disarmed
```

Can't see an 'alarmed' or 'in alarm' state.

Likewise what should I put in this node red value for when in alarm? 
![Screen Shot 2022-07-20 at 1 24 29 pm](https://user-images.githubusercontent.com/14887388/179889784-23c20991-848c-40ba-b498-08a947a43e67.png)

**Comment by samturner3** [comment]
So I found it is a state called 'triggered'.

However it is not at all reliable: 

I am trying to trigger a siren when it gets triggered and turn off when disarmed via node red.
It is impossible just by looking at the state changes of `alarm_control_panel.homebase`!!

### Results of my test:
#### Arming with exit delay:
_It goes into `armed_away` -> (exit delay expires) -> `triggered` -> `armed_away`_
![Screen Shot 2022-07-20 at 2 19 41 pm](https://user-images.githubusercontent.com/14887388/179896023-d69171e0-28fd-4901-975e-f0f5212a06b9.png)
I don't understand why in this scenario it goes to triggered for a second??

#### Open door (trigger alarm) while armed:
It goes into `triggered` -> `armed_away`_ (even while alarm is sounding!) See timestamps.
![Screen Shot 2022-07-20 at 2 20 50 pm](https://user-images.githubusercontent.com/14887388/179896152-69c74d1b-d8e5-40a1-be6d-c0f726f258f5.png)
I don't understand why in this scenario it only goes to triggered for a second, then armed_away again even while the alarm is sounding on the homebase??

#### Disarm while in alarm (triggered):
_Note: It goes into `triggered` -> `disarmed`_
![Screen Shot 2022-07-20 at 2 21 02 pm](https://user-images.githubusercontent.com/14887388/179896166-44380461-b029-49d3-803a-0dfdbb0b65ea.png)


So I can't see a solution to reliably trigger the siren on intrusion and turn off on disarm, since when arming it goes from ``triggered` -> `armed_away``, EXACTLY like intrusion, (`triggered` -> `armed_away`)

One can not simply rely on the `triggered` state as it will go off for a second when arming, also only for a second on intruder when it switches back to `armed_away` instantly. 

Any solutions would be greatly appreciated.

**Comment by fuatakgun** [comment]
It goes back to "what is a state" and "what is an event" :)
Station is in a state home - away - disarmed etc.
When it is triggered with a motion, it generates an event but the state is still same, let's say armed.
When p2p event is received, I am just switching the home assistant state from armed to triggered and reverting back.

**Comment by fuatakgun** [comment]
should be fixed in Alpha version

**Reply by samturner3** [reply]
Hi did this go out in v6 release? I'll have to update and try this automation again.
What has changed to solve keeping in mind above ☝️ ?
